colrev/ops/built_in/load_conversion/ris_rispy_loader.py

```diff
@@ -32,82 +33,101 @@ class RisRispyLoader(JsonSchemaMixin):
             package_type=colrev.env.package_manager.PackageEndpointType.search_source,
             selected_packages=[source.get_dict()],
             operation=load_operation,
             ignore_not_available=False,
         )
         endpoint = endpoint_dict[source.endpoint]
-
         records = endpoint.load_fixes(  # type: ignore
             load_operation, source=source, records=records
         )
         return records
 
-    def __parse_ris_file_and_convert_to_bib(
-        self, source: colrev.settings.SearchSource
-    ) -> dict:
-        """Parses given ris file with rispy and convert it to bib record"""
-
-        def add_entry(_entry: dict, key: str, _record: dict) -> None:
-            if key in _entry:
-                _record[key] = _entry[key]
+    def __get_entry_id(self, entry: dict) -> typing.Tuple[str, str, int]:
+        try:
+            authors = entry["first_authors"] + entry["secondary_authors"]
+        except KeyError:
+            authors = entry["authors"]
+        first_three_authors = authors
+        if len(authors) >= 3:
+            first_three_authors = authors[:3]
+        try:
+            year = entry["year"].strip("/")
+        except KeyError:
+            year = entry["publication_year"].strip("/")
+        three_author = "".join(
+            [author.rsplit(",")[0].strip() for author in first_three_authors]
+        )
+        _id = f"{three_author.lower()}{year}"
+        return _id, authors, year
 
-        with open(source.filename, encoding="utf-8") as ris_file:
-            entries = rispy.load(file=ris_file)
+    def __add_entry(self, _entry: dict, key: str, record: dict) -> None:
+        if key in _entry:
+            val = _entry[key]
+            target_key = key if key not in self.replace_with else self.replace_with[key]
+            record[target_key] = val
 
+    def __parse_ris_file_and_convert_to_bib(self) -> dict | None:
+        """Parses given ris file with rispy and convert it to bib record"""
+        if not self.entries:
+            return None
         records = {}
-        for entry in entries:
-            print(entry)
+        for entry in self.entries:
+            entry = dict(sorted(entry.items()))
             # colrev uses first 3 auther + year format
-            try:
-                authors = entry["authors"]
-            except KeyError:
-                authors = entry["first_authors"] + entry["secondary_authors"]
-            first_three_authors = authors
-            if len(authors) >= 3:
-                first_three_authors = authors[:3]
-            try:
-                year = entry["year"]
-            except KeyError:
-                year = entry["publication_year"].strip("/")
-            three_author = "".join(
-                [author.rsplit(",")[0].strip() for author in first_three_authors]
-            )
-            _id = f"{three_author.lower()}{year}"
+            _id, authors, year = self.__get_entry_id(entry)
```

Comment:
The method could be split (it returns more than just the id).

Reply:
Okay